### PR TITLE
Fix linkage errors, qualifier handling, NPE in metrics processing, and revert Xerces upgrade

### PR DESCRIPTION
**Pull request by SmithJosh**
#### What does this PR do?
Fixes a few bugs identified during testing on a downstream project. In particular:
* Fixes a couple of linkage errors encountered when using SOAP endpoints
* Fixes an assumption in code that a qualifier on a query always indicates a search for a derived resource. There is at least one known use case where this is not true.
* Fixes an NPE in pre-query metrics processing
* Reverts a Xerces upgrade as it was causing WSDL policy injection to fail for secure SOAP endpoints
 
#### Who is reviewing it? 
@stustison @paouelle @blen-desta @bakejeyner @garrettfreibott 

#### What are the relevant tickets?
Fixes: https://github.com/codice/ddf/issues/6168

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by SmithJosh** [comment]
build now

File: distribution/ddf-common/src/main/resources/etc/jre.properties [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+ org.w3c.dom.views, \
+ org.w3c.dom.xpath, \
+ org.xml.sax, \
+ org.xml.sax.ext, \
+ org.xml.sax.helpers, \
+ com.sun.nio.sctp
+
+jre-1.8= \
+ javax.accessibility, \
+ javax.activation;version="1.1", \
```
Comment (SmithJosh) [inline distribution/ddf-common/src/main/resources/etc/jre.properties:345]: Added this line

File: distribution/ddf-common/src/main/resources/etc/jre.properties [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+ javax.xml.crypto, \
+ javax.xml.crypto.dom, \
+ javax.xml.crypto.dsig, \
+ javax.xml.crypto.dsig.dom, \
+ javax.xml.crypto.dsig.keyinfo, \
+ javax.xml.crypto.dsig.spec, \
+ javax.xml.datatype, \
+ javax.xml.namespace, \
+ javax.xml.parsers, \
+ javax.xml.soap;version="1.3", \
```
Comment (SmithJosh) [inline distribution/ddf-common/src/main/resources/etc/jre.properties:440]: And added this line

File: catalog/core/catalog-core-metricsplugin/src/main/java/ddf/catalog/metrics/CatalogMetrics.java [hunk 3]
```diff
@@ -95,8 +97,10 @@ public QueryRequest process(QueryRequest input)
       throws PluginExecutionException, StopProcessingException {
     final String scope = getQueryScope(input);
     QueryTypeFilterDelegate queryType = new QueryTypeFilterDelegate();
-    Set<String> sourceIds = input.getSourceIds();
-    if (sourceIds != null) {
+
+    if (input.getSourceIds() != null) {
```
Comment (clockard) [inline catalog/core/catalog-core-metricsplugin/src/main/java/ddf/catalog/metrics/CatalogMetrics.java:101]: Can we update unit tests to check this?

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java [hunk 4]
```diff
@@ -429,8 +430,17 @@ private static URI getResourceOrDerivedUri(ResourceInfo resourceInfo, ResourceRe
...
+
+    // no qualifier means we are retrieving the resource
+    // presence of a qualifier can mean different things. We interpret it based on protocol:
+    //   - content: this always means we are retrieving a derived resource.
+    //   - http(s): we assume the intent is to retrieve a derived resource. Is this always true?
+    //   - other: we assume we are retrieving the resource. This assumption is based on experience
+    //            with protocols that use the qualifier for purposes other than derived resources.
+    //            If protocols besides content and http(s) need to support derived resources, this
+    //            could be an issue.
+    if (qualifier == null || !Arrays.asList("http", "https", "content").contains(scheme)) {
```
Comment (clockard) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java:443]: Update unit tests?

File: platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java [hunk 5]
```diff
@@ -326,7 +326,7 @@ public void handleMessage(Message message) throws Fault {
       }
 
       Collection<AssertionInfo> ais = getAllAssertionsByLocalname(aim, SPConstants.ISSUED_TOKEN);
-      if (ais == null) {
+      if (ais == null || ais.isEmpty()) {
```
Comment (clockard) [inline platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java:329]: Update unit tests?

File: catalog/core/catalog-core-metricsplugin/src/main/java/ddf/catalog/metrics/CatalogMetrics.java (hunk 3 above)
Comment (SmithJosh) [reply catalog/core/catalog-core-metricsplugin/src/main/java/ddf/catalog/metrics/CatalogMetrics.java:101]: Added a unit test for this

File: platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java (hunk 5 above)
Comment (SmithJosh) [reply platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java:329]: There are no unit tests to update for this 😕

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java (hunk 4 above)
Comment (SmithJosh) [reply catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java:443]: Hm... same here. No unit tests for this derived resource handling logic. I can try to add something for this change, but the whole method needs tests.

Comment (SmithJosh) [reply catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java:445]: Added unit tests

**Comment by SmithJosh** [comment]
build now

**Pull request merged**